### PR TITLE
Add LoginHint option for InteractiveBrowserCredential

### LINKED ISSUE #15599: login_hint support for InteractiveBrowserCredential
Seems sensible to finish #14920 first and let MSAL handle this but it's uncertain when that will be possible because MSAL doesn't yet allow setting login_hint (https://github.com/AzureAD/microsoft-authentication-library-for-go/issues/271). Adding it to our own auth code flow would create a blocker for  transitioning to MSAL.

### PR DESCRIPTION
**Pull request by chlowell**
Closes #15599

### CODE REVIEW

File: sdk/azidentity/interactive_browser_credential.go [hunk 1]
```diff
@@ -73,11 +77,7 @@ func (c *InteractiveBrowserCredential) GetToken(ctx context.Context, opts policy
 		return azcore.AccessToken{Token: ar.AccessToken, ExpiresOn: ar.ExpiresOn.UTC()}, err
 	}
 
-	o := []public.InteractiveAuthOption{}
-	if c.options.RedirectURL != "" {
-		o = append(o, public.WithRedirectURI(c.options.RedirectURL))
-	}
-	ar, err = c.client.AcquireTokenInteractive(ctx, opts.Scopes, o...)
+	ar, err = c.client.AcquireTokenInteractive(ctx, opts.Scopes, public.WithLoginHint(c.options.LoginHint), public.WithRedirectURI(c.options.RedirectURL))
```
Comment (jhendrixMSFT) [inline sdk/azidentity/interactive_browser_credential.go:80]: If `options.LoginHint` and `options.RedirectURL` are empty, these options are just no-op?

Comment (chlowell) [reply sdk/azidentity/interactive_browser_credential.go:80]: Yes. The values end up as string fields on a struct, so an explicit empty string is the same as not setting these options. Is that weird? I'm happy to keep the conditionals if they prevent future maintainers from wondering about this.

Comment (jhendrixMSFT) [reply sdk/azidentity/interactive_browser_credential.go:80]: I think it's fine and makes for simpler code on our side, assuming the contract for these APIs doesn't change on the MSAL side (seems unlikely and would show up in our unit testing).

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**